server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -25,7 +26,37 @@ package object server {
   /**
    * A [[Service]] that returns an optional result.
    */
-  type PartialService[A, B] = Service[A, Option[B]]
+  type PartialService[A, B] = Kleisli[({type L[x] = OptionT[Task, x]})#L, A, B]
```

Comment:
fwiw : I tend to write type aliases instead of type lambdas (`private type OTask[x] = OptionT[Task, x]`)